### ISSUE TITLE
Add support for "default verb"

### ISSUE THREAD

**Issue by notofug** [dotnet/command-line-api#2325]
[CommandLineParser](https://github.com/commandlineparser/commandline) supports ["default verb"](
https://github.com/commandlineparser/commandline/wiki/Verbs#default-verb) which is a nice feature.

Can this be added to System.CommandLine or is there a way to achive same effect already?

**Comment by notofug** [comment]
Yes.. but any options/arguments added to root-command seem to extend to all 'sub-commands' as well (--help will list them also for any subcommand since they are at root)
Maybe that can be worked around also, but I dont see an obvious way

**Comment by notofug** [comment]
> Similar control is not available for arguments, though.

Right ; this was my use-case and what I ran into

Anyway; thank you guys for advice and having this in mind as you decide how things progress 👍

**Comment by KathleenDollard** [comment]
I'm not sure I understand. Is it that you want these to be equivalent:

```c#
app -a -b
app foo -a -b
```

while, not allowing -a or -b if a different subcommand is used? IOW, not allowing:

```c#
app -a -b bar
```

I am curious as to when this would be helpful and whether there are any commonly used CLIs that allow this. I think it would often obscure intent.

If we had ubiquitous tab completion (I am a dreamer), would it still be important? IOW, is it just to avoid typing or is there another reason?

**Comment by notofug** [comment]
Yes what I was looking for was a way to be able to say :
`app -a -b`
and 
`app bar -c`

As it is now I believe the -a -b would unconditionally/automatically apply also to subcommand 'bar' (which is unwanted)

Anyway, I moved on and you are welcome to close or down-prioritize this request as nothing is suffering from it really

**Comment by BurningEnlightenment** [comment]
My use case for this functionality is a small devtool created for a singular task. It also offers a `config` subcommand (think of it like `git config`). The main task is used frequently (~2 per workday) while my coworkes rarely need to adjust the `config`. Furthermore the executable is aptly named and repeating the taskname or naming the subcommand `do` feels… wrong?

**Comment by fourpastmidnight** [comment]
Doesn't this _kind of_ go away with the release of v2.0.0? Now that parsing and invocation are separate from one another? One can parse the command line, and you don't _need_ to specify any actions for commands, but you could take the `ParseResult` and create your own logic around it. So, the root command could accept the same arguments as the "default command", and, if your parse result contains `app -a -b bar -x`, then you could handle that with your own logic.

Alternatively, you may _still_ be able to provide actions for your commands, but you would still need to validate the `ParseResult` to ensure that if any arguments are given to the root command, that if also any subcommands exist in the `ParseResult`, you'd take whatever action necessary to alert the user of their mistake.
